### PR TITLE
perf(musa): partition mixed Qwen seeded sampling

### PR DESCRIPTION
**Pull request by yeahdongcn**
## Summary

This is a dependent follow-up to draft PR #123:
https://github.com/MooreThreads/vllm-musa/pull/123

It extends the default-off Qwen legacy Gumbel experiment to batches that mix
explicitly seeded and unseeded requests. The implementation keeps the existing
per-row multinomial behavior for unseeded rows, uses the batched stateless
Gumbel path only for seeded rows, and restores the original row order.

The PR is intentionally stacked on
`perf/musa-60043-qwen-family-formal`; it does not change vLLM-Omni and does not
enable the experiment by default.

## Implementation

- Accept a non-empty sparse generator map whose keys are valid batch rows.
- Apply the existing top-k filter once to the full batch.
- Compact seeded rows for one stateless Gumbel launch, remapping generator
  rows only inside that launch.
- Sample unseeded rows with the existing multinomial helper and global MUSA RNG.
- Scatter both results back to the original batch order.
- Preserve the all-seeded path from PR #123 without adding partition/index
  launches.
- Add CPU-state unit coverage for sparse validation, row restoration, seed and
  offset handoff, and the unseeded probability path.

### Fixed-logits semantic/operator probe

One isolated S5000, BS 1/4/16/64:

- all seeded rows matched a standalone Gumbel launch for the same seed/offset;
- unseeded rows matched the all-multinomial control exactly;
- the next global MUSA RNG token matched control;
- every seeded generator advanced from offset 0 to 4.

BS64 fixed-filtered-logits host-timed comparison:

| Seeded rows | Existing multinomial | Partitioned candidate | Speedup |
|---:|---:|---:|---:|
| 25% | 43.498 ms | 35.062 ms | 1.241x |
| 50% | 44.150 ms | 24.206 ms | 1.824x |
| 75% | 42.981 ms | 14.455 ms | 2.974x |
| 100% | 45.195 ms | 4.466 ms | 10.119x |

This is operator evidence, not an end-to-end claim.

Raw result: `generated/MUSA-60043/probes/legacy-mixed-partition-v1.json`

Archive SHA256:
`9d44333d67da46d8efa6a84272ef12f83f02335a736b62cb3dee82f6a8f42df0`

### Compiled serving smoke

Qwen3.5-27B BF16 TP2, normal `VLLM_COMPILE` /
`FULL_AND_PIECEWISE` / FA3 path, with two seeded and two unseeded concurrent
requests. Runtime diagnostics observed `rows=4`, `generators=[0, 1]`, and the
mixed partition marker; all four requests returned HTTP 200.

Serving archive SHA256:
`37ef54631c411183da542d1d4e43805c6d338b692658287f73ade78c44799b84`

No formal mixed-composition serving A/B is claimed yet. That is the next gate,
after review of this default-off prototype.

## Validation

- `python -m py_compile vllm_musa/v1/sample/topk_topp_sampler.py tests/test_sampler_cpu_state_fast_paths.py`
- `ruff check vllm_musa/v1/sample/topk_topp_sampler.py tests/test_sampler_cpu_state_fast_paths.py`
- `ruff format --check vllm_musa/v1/sample/topk_topp_sampler.py tests/test_sampler_cpu_state_fast_paths.py`
- `git diff --check`
- Remote S5000 semantic/operator probe and compiled serving smoke: passed.
- Local pytest collection: not run to completion because the host Python
  environment lacks `numpy`; the remote probe and the existing PR #123 test
  environment provide hardware-path coverage.

## Risk / rollout

The environment gate remains `VLLM_MUSA_QWEN_LEGACY_GUMBEL=0` by default.
Speculative decoding, logprobs, deferred min-p, unsupported vocabularies, and
invalid generator state continue to fall back to the existing sampler.

### CODE REVIEW

**Comment by yeahdongcn** [comment]
Superseded by #138, which consolidates this optimization into one automatic Qwen fast-path implementation with exact-head performance and regression evidence and no manual feature switch. Closing this Draft to keep a single review and merge surface. Please review #138.
